lens-cube/src/main/java/org/apache/lens/cube/parse/StorageTableResolver.java

```diff
@@ -269,56 +273,52 @@ private void resolveStorageTable(CubeQueryContext cubeql) throws LensException {
       }
 
       boolean valid = false;
-      Set<CandidateTablePruneCause.CandidateTablePruneCode> codes = new HashSet<>();
+      // There could be multiple causes for the same time range.
+      Set<CandidateTablePruneCause.CandidateTablePruneCode> pruningCauses = new HashSet<>();
       for (TimeRange range : cubeql.getTimeRanges()) {
         boolean columnInRange = client
           .isStorageTableCandidateForRange(storageTable, range.getFromDate(), range.getToDate());
-        boolean partitionColumnExists = client.partColExists(storageTable, range.getPartitionColumn());
-        valid = columnInRange && partitionColumnExists;
-        if (valid) {
-          break;
-        }
         if (!columnInRange) {
-          codes.add(TIME_RANGE_NOT_ANSWERABLE);
+          pruningCauses.add(TIME_RANGE_NOT_ANSWERABLE);
           continue;
         }
-        // This means fallback is required.
+        boolean partitionColumnExists = client.partColExists(storageTable, range.getPartitionColumn());
+        valid = partitionColumnExists;
         if (!partitionColumnExists) {
           String timeDim = cubeql.getBaseCube().getTimeDimOfPartitionColumn(range.getPartitionColumn());
-          if (!sc.getFact().getColumns().contains(timeDim)) {
-            // Not a time dimension so no fallback required.
-            codes.add(TIMEDIM_NOT_SUPPORTED);
-            continue;
-          }
+          //          if (!sc.getFact().getColumns().contains(timeDim)) {
+          //           // Not a time dimension so no fallback required.
+          //          pruningCauses.add(TIMEDIM_NOT_SUPPORTED);
+          //        continue;
+          //       }
           TimeRange fallBackRange = getFallbackRange(range, sc.getFact().getCubeName(), cubeql);
           if (fallBackRange == null) {
             log.info("No partitions for range:{}. fallback range: {}", range, fallBackRange);
+            pruningCauses.add(TIME_RANGE_NOT_ANSWERABLE);
             continue;
           }
           valid = client
             .isStorageTableCandidateForRange(storageTable, fallBackRange.getFromDate(), fallBackRange.getToDate());
-          if (valid) {
-            break;
-          } else {
-            codes.add(TIME_RANGE_NOT_ANSWERABLE);
+          if (!valid) {
+            pruningCauses.add(TIME_RANGE_NOT_ANSWERABLE);
           }
         }
       }
       if (!valid) {
```

Comment:
@lavkesh This seems wrong. What if a SC is valid for TimeRange1 and invalid for TimeRange2. In this case we should not prune .